components/scream/src/physics/shoc/atmosphere_macrophysics.hpp

```diff
@@ -28,16 +28,17 @@ class SHOCMacrophysics : public scream::AtmosphereProcess
   using C            = physics::Constants<Real>;
   using KT           = ekat::KokkosTypes<DefaultDevice>;
 
-  using Spack = typename SHF::Spack;
-  using IntSmallPack = typename SHF::IntSmallPack;
-  using Smask = typename SHF::Smask;
-  using view_1d  = typename SHF::view_1d<Real>;
-  using view_1d_const  = typename SHF::view_1d<const Real>;
-  using view_2d  = typename SHF::view_2d<SHF::Spack>;
-  using view_2d_const  = typename SHF::view_2d<const Spack>;
-  using sview_2d = typename KokkosTypes<DefaultDevice>::template view_2d<Real>;
-  using view_3d = typename SHF::view_3d<Spack>;
-  using view_3d_const = typename SHF::view_3d<const Spack>;
+  using Spack                = typename SHF::Spack;
+  using IntSmallPack         = typename SHF::IntSmallPack;
+  using Smask                = typename SHF::Smask;
+  using view_1d              = typename SHF::view_1d<Real>;
+  using view_1d_const        = typename SHF::view_1d<const Real>;
+  using view_1d_const_double = typename SHF::view_1d<const double>;
+  using view_2d              = typename SHF::view_2d<SHF::Spack>;
+  using view_2d_const        = typename SHF::view_2d<const Spack>;
+  using sview_2d             = typename KokkosTypes<DefaultDevice>::template view_2d<Real>;
+  using view_3d              = typename SHF::view_3d<Spack>;
+  using view_3d_const        = typename SHF::view_3d<const Spack>;
 
```

Comment:
Not a complaint about this PR, but just commenting on something that makes me a bit uncomfortable: SHF usually means sensible heat flux to atm scientists. It looks like this is being used to denote something very different here, which will confuse atm people reading the code. Also, I'm personally not clear what SHF means here.

Reply:
I'm guessing ShocFunctions?

Reply:
I think a lot of types are duplicated between, say, P3Functions and ShocFunctions. I am wondering whether a `Types` struct in `src/share/scream_types.hpp` might be beneficial to avoid redundancies, and also to have somewhat more intuitive names, like

```
using view_3d = typename Types::view_3d<Spack>;
```

Reply:
Yeah, I think this is confusing, but it's used all over shoc. I wouldn't mind implementing this shared types struct. Or at the least maybe use a more discriptive name (`SHOCFunc` or even just `SHOC`).

Reply:
Yeah, I agree a Types struct would be nice. Seems like something we should work on for SCREAMv1.1 rather than changing what we've already done now?

Reply:
I already changed my mind, in the sense that P3, SHOC, and whatnot might have slightly different needs, so maybe a common types struct won't give us much. However, on a slightly related topic, I advocated in the past for splitting P3Functions into a P3Types struct and a functions namespace (`Types` is only 3 chars less then `Functions`, so not big saving probably, but there were other reasons). One day I'll bring it up again to haunt you with philosophical discussions on C++ and compilation units...